src/App.tsx

```diff
@@ -1,24 +1,78 @@
 import React from 'react';
 import './App.scss';
+import { Clock } from './components/Clock';
 
-const App: React.FC = () => {
-  const timerId: NodeJS.Timer = setInterval(() => {
-    const date: Date = new Date();
-
-    // eslint-disable-next-line
-    console.log(date.toLocaleTimeString());
-  }, 1000);
-
-  return (
-    <div className="App">
-      <h1>React clock</h1>
-      <p>
-        Current time:
-        {' '}
-        {/* Print the time here instead of DevTools */}
-      </p>
-    </div>
-  );
-};
+class App extends React.Component {
+  state = {
+    timerId: undefined,
+    currentTime: new Date().toLocaleTimeString(),
+    isClockVisible: true,
+  };
+
+  componentDidMount() {
+    this.startClockTimer();
+  }
+
+  componentDidUpdate() {
+    if (this.state.isClockVisible && !this.state.timerId) {
+      this.startClockTimer();
+    }
+  }
+
+  startClockTimer = () => {
+    const id = setInterval(() => {
+      const time = new Date().toLocaleTimeString();
+
+      // eslint-disable-next-line
+      console.log(time);
+
+      this.setState({
+        timerId: id,
+        currentTime: time,
+      });
+    }, 1000);
+  };
+
+  render(): React.ReactNode {
+    const {
+      currentTime,
+    } = this.state;
+
+    return (
+      <div className="App">
+        <h1>React clock</h1>
+
+        {this.state.isClockVisible
+        && (
+          <Clock
+            time={currentTime}
+            id={this.state.timerId}
+          />
+        )}
+
+        <div className="App__buttons">
+          <button
+            type="button"
+            onClick={() => this.setState({ isClockVisible: true })}
```

Comment:
do not make these functions anonymous. define them outside the render method and give them proper names